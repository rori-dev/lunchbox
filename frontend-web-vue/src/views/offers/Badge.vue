<template>
  <div
    class="rounded-full
           font-light text-neutral-900
           px-3 py-1 mr-1"
    :class="props.color"
  >
    {{ props.label }}
  </div>
</template>

<script lang="ts">
import { defineComponent } from 'vue'
export default defineComponent({ name: 'Badge' })
</script>

<script setup lang="ts">
import { defineProps } from 'vue'

const props = defineProps<{
  label: string
  color: string
}>()
</script>
