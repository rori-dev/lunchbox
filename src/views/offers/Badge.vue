<template>
  <div
    class="rounded-full
            bg-yellow-200 text-gray-900 font-light
            px-3 py-1 mr-1"
  >
    {{ label }}
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'

@Component
export default class Badge extends Vue {
  @Prop() label!: string
}
</script>
