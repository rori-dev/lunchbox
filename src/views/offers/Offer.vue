<template>
  <div
    class="flex flex-col
           w-full pt-4"
  >
    <div class="flex items-baseline w-full">
      <span class="flex-grow text-xl">{{ offer.name }}</span>
      <span class="text-lg font-light pl-2">â‚¬</span>
      <span class="text-xl pl-2">{{ offer.price | formatEuro }}</span>
    </div>
    <span
      class="hidden sm:block
               pl-1 text-gray-900 font-light"
    >
      {{ offer.description }}
    </span>
    <div class="hidden sm:flex pt-1">
      <Badge v-for="tagLabel in offer.tags" :key="tagLabel" :label="tagLabel" />
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'
import { LunchOffer } from '@/model'
import Badge from '@/views/offers/Badge.vue'

@Component({
  components: {
    Badge,
  },
})
export default class Offer extends Vue {
  @Prop() offer!: LunchOffer
}
</script>
