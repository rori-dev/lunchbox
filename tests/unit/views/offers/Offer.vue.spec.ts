import Offer from '@/views/offers/Offer.vue'
import { gyros } from '@tests/unit/test-data'
import { mountUnit } from '@tests/unit/test-util'
import Badge from '@/views/offers/Badge.vue'

describe('Offer', () => {
  it('renders snapshot', () => {
    const wrapper = mountUnit(Offer, {
      offer: gyros,
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders name, description & tags', () => {
    const wrapper = mountUnit(Offer, {
      offer: gyros,
    })

    expect(wrapper.text()).toContain(gyros.name)
    expect(wrapper.text()).toContain(gyros.description)
    const items = wrapper.findAll(Badge)
    expect(items.length).toEqual(2)
    expect(items.at(0).props('label')).toEqual(gyros.tags[0])
    expect(items.at(1).props('label')).toEqual(gyros.tags[1])
  })
})
