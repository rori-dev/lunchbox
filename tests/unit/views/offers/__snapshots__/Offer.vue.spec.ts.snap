// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Offer renders snapshot 1`] = `
<div
  class="flex flex-col
         w-full pt-4"
>
  <div
    class="flex items-baseline w-full"
  >
    <span
      class="flex-grow text-xl"
    >
      Gyros
    </span>
     
    <span
      class="text-lg font-light pl-2"
    >
      â‚¬
    </span>
     
    <span
      class="text-xl pl-2"
    >
      3,50
    </span>
  </div>
   
  <span
    class="hidden sm:block
             pl-1 text-gray-900 font-light"
  >
    
    mit Pommes
  
  </span>
   
  <div
    class="hidden sm:flex pt-1"
  >
    <badge-stub
      label="Tagesgericht"
    />
    <badge-stub
      label="lecker"
    />
  </div>
</div>
`;
